extensions/amp-google-assistant-assistjs/0.1/assistjs-config-service.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The shared service takes assist.js related config from <amp-google-assistant-assistjs-config>
+ * and provides config parameters to other assist.js components.
+ */
+
+export class AssistjsConfigService {
+  /** */
+  constructor() {
+    const config = JSON.parse(
+      document.getElementsByTagName('amp-google-assistant-assistjs-config')[0]
```

Comment:
A few notes here:
1. We should assert there aren't multiple of these tags on the page.
2. I'm not sure if we are guaranteed the DOM has finished loading at the point of service construction. I wouldn't block on this right now, but depending on if this is safe or not we may need to move this parsing around to ensure it occurs after dom load.

Reply:
1. assertion added
2. added a promise to wait for in config service. When widgets tries to get iframe url from config service, it would first wait for the promise to got resolved, which would be resolved when the json is retrieved from the config element.